File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        Map<String, HashSet<String>> reportedID = new HashMap<>(); // key:신고당한사람, value:신고한사람
        Map<String, Integer> idIndex = new HashMap<>(); // key:id, value:id의 인덱스
        for(int i = 0; i < id_list.length; i++) {
            idIndex.put(id_list[i], i);
        }
        for(int i = 0; i < report.length; i++) {
            String[] cur = report[i].split(" ");
            reportedID.computeIfAbsent(cur[1], k1 -> new HashSet<>());
            reportedID.get(cur[1]).add(cur[0]);
        }
        for(int i = 0; i < id_list.length; i++) {
            Set<String> reporter = reportedID.get(id_list[i]);
            if(reporter != null && reporter.size() >= k) { // 회원 정지
                for(String s : reporter) {
                    answer[idIndex.get(s)]++;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.kt
class Solution {
    fun solution(id_list: Array<String>, report: Array<String>, k: Int): IntArray {
        var answer = IntArray(id_list.size) { 0 }
        val reportedID = mutableMapOf<String, HashSet<String>>()
        val idIndex = mutableMapOf<String, Int>()

        id_list.forEachIndexed{ idx, id ->
            idIndex[id] = idx
        }
        for(reporting in report) {
            val token = reporting.split(" ")
            reportedID.putIfAbsent(token[1], HashSet())
            reportedID[token[1]]?.add(token[0])
        }
        for(id in id_list) {
            val reporter = reportedID[id]
            if(reporter != null && reporter.size >= k) {
                for(who in reporter) {
                    answer[idIndex[who]!!] += 1
                }
            }
        }
        return answer
    }
}